src/pages/ConnectWallet/components/WalletList.tsx

```diff
@@ -0,0 +1,163 @@
+import { Alert, AlertDescription, AlertIcon, Center, Spinner, Stack } from '@chakra-ui/react'
+import { useQuery } from '@tanstack/react-query'
+import { useCallback, useMemo, useState } from 'react'
+import { useHistory } from 'react-router'
+import { MobileWalletDialogRoutes } from 'components/MobileWalletDialog/types'
+import { WalletActions } from 'context/WalletProvider/actions'
+import { KeyManager } from 'context/WalletProvider/KeyManager'
+import { useLocalWallet } from 'context/WalletProvider/local-wallet'
+import { MobileConfig } from 'context/WalletProvider/MobileWallet/config'
+import { getWallet, listWallets } from 'context/WalletProvider/MobileWallet/mobileMessageHandlers'
+import type { RevocableWallet } from 'context/WalletProvider/MobileWallet/RevocableWallet'
+import { useWallet } from 'hooks/useWallet/useWallet'
+
+import { WalletCard } from './WalletCard'
+
+export type WalletInfo = {
+  id?: string
+  name?: string
+  createdAt?: number
+}
+
+type MobileWalletDialogProps = {
+  footerComponent?: JSX.Element
+  isEditing?: boolean
+}
+
+export const MobileWalletList: React.FC<MobileWalletDialogProps> = ({
+  footerComponent,
+  isEditing,
+}) => {
+  const { dispatch, getAdapter, state } = useWallet()
+  const { walletInfo } = state
+  const localWallet = useLocalWallet()
+  const history = useHistory()
+  const [error, setError] = useState<string | null>(null)
+
+  const { isLoading, data: wallets } = useQuery({
+    queryKey: ['listWallets'],
+    staleTime: 0,
+    gcTime: 0,
+    refetchOnMount: true,
```

Comment:
@NeOMakinG discussed this one out with beard, for context `listWallets` is this weird magic event-based messaging from the mobile app and there are no deps here so no staleTime, no gcTime and refresh on mount instead of the regular react-query business is expected